### ISSUE TITLE
Tensorboard does not match keras loss output for custom model

### ISSUE THREAD

**Issue by jmattdale** [tensorflow/tensorflow#56100]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

binary

### Tensorflow Version

2.6.0

### Custom Code

Yes

### OS Platform and Distribution

Linux Ubuntu 20.04

### Python version

3.8.10

### CUDA/cuDNN version

8204

### Current Behaviour?

```shell
Loss values in callbacks and tensorboard do not match output from loss_dict output in shell.
```

### Standalone code to reproduce the issue

```shell
class LearningRateLogger(tf.keras.callbacks.Callback):
    def on_epoch_end(self, epoch, logs=None):
        logs["learning_rate"] = self.model.optimizer.lr
    def on_batch_end(self, batch, logs=None):
        print('\n loss from on_batch_end callback: {}'.format(logs['loss']))


from subclassed Model:

    def train_step(self, data):
        with tf.GradientTape() as tape:
            y_pred = self(data.images, training=True)
            loss = self.loss_fn(y_pred, data.gt)
        gradients = tape.gradient(loss, self.trainable_variables)
        self.optimizer.apply_gradients(zip(gradients, self.trainable_variables))

        loss_dict = {
            'center_classification_loss': loss,
            'loss': loss
        }
        return loss_dict
```

### Relevant log output

```shell
36/955 [>.............................] - ETA: 11:03 - center_classification_loss: 764.4570 - loss: 764.4570
 loss from on_batch_end callback: 27.19649314880371
 37/955 [>.............................] - ETA: 10:59 - center_classification_loss: 745.5407 - loss: 745.5407
 loss from on_batch_end callback: 64.55301666259766
 38/955 [>.............................] - ETA: 10:56 - center_classification_loss: 727.1075 - loss: 727.1075
 loss from on_batch_end callback: 45.08071517944336
 39/955 [>.............................] - ETA: 10:55 - center_classification_loss: 710.0048 - loss: 710.0048
 loss from on_batch_end callback: 60.10035705566406
 40/955 [>.............................] - ETA: 10:53 - center_classification_loss: 694.1853 - loss: 694.1853
 loss from on_batch_end callback: 77.22644805908203
 41/955 [>.............................] - ETA: 10:51 - center_classification_loss: 678.0594 - loss: 678.0594
 loss from on_batch_end callback: 33.022945404052734
 42/955 [>.............................] - ETA: 10:49 - center_classification_loss: 662.8418 - loss: 662.8418
 loss from on_batch_end callback: 38.92039489746094
 43/955 [>.............................] - ETA: 10:49 - center_classification_loss: 649.2812 - loss: 649.2812
 loss from on_batch_end callback: 79.73693084716797
 44/955 [>.............................] - ETA: 10:46 - center_classification_loss: 636.5957 - loss: 636.5957
 loss from on_batch_end callback: 91.1175765991211
 45/955 [>.............................] - ETA: 10:45 - center_classification_loss: 623.2176 - loss: 623.2176
 loss from on_batch_end callback: 34.58213424682617
 46/955 [>.............................] - ETA: 10:44 - center_classification_loss: 612.0084 - loss: 612.0084
 loss from on_batch_end callback: 107.5921401977539
 47/955 [>.............................] - ETA: 10:43 - center_classification_loss: 600.4400 - loss: 600.4400
 loss from on_batch_end callback: 68.297607421875
 48/955 [>.............................] - ETA: 10:40 - center_classification_loss: 589.0859 - loss: 589.0859
 loss from on_batch_end callback: 55.439815521240234
 49/955 [>.............................] - ETA: 10:37 - center_classification_loss: 578.3141 - loss: 578.3141
 loss from on_batch_end callback: 61.2695426940918
 50/955 [>.............................] - ETA: 10:36 - center_classification_loss: 567.3053 - loss: 567.3053
 loss from on_batch_end callback: 27.87373924255371
 51/955 [>.............................] - ETA: 10:36 - center_classification_loss: 559.1129 - loss: 559.1129
 loss from on_batch_end callback: 149.49432373046875
 52/955 [>.............................] - ETA: 10:34 - center_classification_loss: 550.9732 - loss: 550.9732
 loss from on_batch_end callback: 135.84913635253906
 53/955 [>.............................] - ETA: 10:31 - center_classification_loss: 541.7439 - loss: 541.7439
 loss from on_batch_end callback: 61.8183479309082
 54/955 [>.............................] - ETA: 10:29 - center_classification_loss: 532.4198 - loss: 532.4198
```
</details>

**Comment by nochichi** [comment]
I may be facing the same problem.

I have not confirmed the value of the tensor board, but I think that it is not learning correctly.

In my case, the value of loss gradually increases to a very large value, eventually becoming nan.

Until the day before yesterday, the same code worked correctly.

**Comment by sushreebarsa** [comment]
@jmattdale 
Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues)
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999)
Thank you!
